tiledb/sm/query/readers/sparse_index_reader_base.h

```diff
@@ -296,11 +296,72 @@ class SparseIndexReaderBase : public ReaderBase {
    * it is really required to determine if a query is incomplete from the client
    * side of a cloud request.
    */
-  struct ReadState {
+  class ReadState {
+   public:
+    /* ********************************* */
+    /*     CONSTRUCTORS & DESTRUCTORS    */
+    /* ********************************* */
+
+    ReadState() = delete;
+
+    ReadState(size_t frag_idxs_len)
+        : frag_idx_(frag_idxs_len) {
+    }
+
+    ReadState(std::vector<FragIdx>&& frag_idx, bool done_adding_result_tiles)
+        : frag_idx_(std::move(frag_idx))
+        , done_adding_result_tiles_(done_adding_result_tiles) {
+    }
+
+    /* ********************************* */
+    /*                API                */
+    /* ********************************* */
+
+    /**
+     * Return whether the tiles that will be processed are loaded in memory.
+     * @return Done adding result tiles.
+     */
+    const bool& done_adding_result_tiles() const {
+      return done_adding_result_tiles_;
+    }
+
+    /**
+     * Sets the flag that determines whether the tiles that will be processed
+     * are loaded in memory.
+     * @param done_adding_result_tiles Done adding result tiles.
+     * @return void
+     */
+    inline void set_done_adding_result_tiles(bool done_adding_result_tiles) {
```

Comment:
This should only be changed from false to true... Remove the parameter.

Reply:
This isn't true--this code is in sparse_index_reader_base.cc.
```c++
Status SparseIndexReaderBase::load_initial_data() {
  if (initial_data_loaded_) {
    return Status::Ok();
  }

  auto timer_se = stats_->start_timer("load_initial_data");
  read_state_.set_done_adding_result_tiles(false);
```
Additionally, there is code where this occurs in sparse_global_order_reader.cc and sparse_unordered_with_dups_reader.cc:
```c++
read_state_.set_done_adding_result_tiles(done_adding_result_tiles);
```